cartridge/auth.lua

```diff
@@ -615,42 +621,19 @@ local function remove_user(username)
     end)
 end
 
-local _response_mt = {
-    __index = {
-        finalize = function(self, resp)
-            for k, v in pairs(resp) do
-                if type(self[k]) == 'table'
-                and type(v) == 'table'
-                then
-                    -- merge tables
-                    for tk, tv in pairs(v) do
-                        self[k][tk] = tv
-                    end
-                else
-                    self[k] = v
-                end
-            end
-
-            return self
-        end
-    },
-}
 --- Authorize an HTTP request.
 --
--- Try to get username from cookies or basic HTTP authentication.
+-- Get username from cookies or basic HTTP authentication.
 --
--- @function check_request
+-- (**Added** in v1.1.0-4)
+-- @function authorize_request
 -- @within Authorizarion
--- @param req An HTTP request
+-- @tparam table request
 -- @treturn boolean Access granted
--- @treturn table HTTP response template
-local function check_request(req)
-    local fiber_storage = fiber.self().storage
-    -- clean fiber storage to behave correctly
-    -- when user logouts within keepalive session
-    fiber_storage['auth_session_username'] = nil
-
-    local resp = setmetatable({}, _response_mt)
+local function authorize_request(req)
+    checks('table')
+    local http_session = {}
+    fiber.self().storage['http_session'] = http_session
```

Comment:
I suggest to hard-wipe fiber.storage for all requrests, via
httpd.before_dispatch hook